### PR TITLE
feat: relay id data photos to server

### PR DESCRIPTION
**Pull request by josepfo**

### CODE REVIEW

**Comment by josepfo** [comment]
[Capture and relay ID Data stills to operator server if not parsed](https://trello.com/c/U7bZ7bNI/607-capture-and-relay-id-data-stills-to-operator-server-if-not-parsed)

File: lib/customers.js [hunk 1]
```diff
@@ -569,6 +570,77 @@ function updatePhotoCard (id, patch) {
     })
 }
 
+/**
+ * @param {String} imageData image encoded
+ * @param {String} directory directory path of id card data for a certain user
+ */
+
```
Comment (chaotixkilla) [inline lib/customers.js:578]: An extra newline here.

Normally, I would say to remove the comments, but as it is a function which touches on lower level stuff, I like the extra comments

**Review by chaotixkilla — APPROVED** [review]

File: lamassu-remote-install/install [hunk 2]
```diff
@@ -19,6 +19,7 @@ BLOCKCHAIN_DIR=/mnt/blockchains
 OFAC_DATA_DIR=/var/lamassu/ofac
 ID_PHOTO_CARD_DIR=/opt/lamassu-server/idphotocard
 FRONTCAMERA_DIR=/opt/lamassu-server/frontcamera
+OPERATOR_DIR=/opt/lamassu-server/operatordata
```
Comment (RafaelTaranto) [inline lamassu-remote-install/install:22]: Hmm, we might need to add this to the lamassu-install repo as well.

File: lib/customers.js [hunk 3]
```diff
@@ -569,6 +570,63 @@ function updatePhotoCard (id, patch) {
...
+function updatePhotos (imagesData, id, dir) {
+  return Promise.resolve(imagesData)
+    .then(patch => {
+      if (_.isEmpty(imagesData)) {
+        return patch
+      }
+      const newPatch = {}
+      // i.e. ../<lamassu-server-home>/<operatorid>/<customerid>/idcarddata
+      console.log(dir, 'directoryyyy')
```
Comment (RafaelTaranto) [inline lib/customers.js]: console.logs

File: lib/customers.js [hunk 4]
```diff
@@ -569,6 +570,63 @@ function updatePhotoCard (id, patch) {
...
+      const dirname = path.join(dir)
+      console.log(dirname, 'dirnaaaaaaaaaaaaaaame')
```
Comment (RafaelTaranto) [inline lib/customers.js]: console.log

File: lib/customers.js [hunk 5]
```diff
@@ -569,6 +570,63 @@ function updatePhotoCard (id, patch) {
...
+      // create the directory tree if needed
+      _.attempt(() => makeDir.sync(dirname))
+      const promises = imagesData.map((imageData, index) => {
+        console.log(index)
```
Comment (RafaelTaranto) [inline lib/customers.js]: another one

File: lib/customers.js [hunk 6]
```diff
@@ -569,6 +570,63 @@ function updatePhotoCard (id, patch) {
...
+/**
+ * @param {String} id customer id
+ * @param {Object} patch customer latest id card photos
+ * @returns {Promise<Object>} new patch to be applied
+ */
+function updateIdCardData (patch, id) {
+  /* TODO: fetch operator id */
+  const operatorId = 'id-operator'
+  const directory = operatorDataDir + '/' + operatorId + '/' + id + '/'
+  console.log(directory)
```
Comment (RafaelTaranto) [inline lib/customers.js]: another one

File: lib/routes/customerRoutes.js [hunk 7]
```diff
@@ -90,5 +103,6 @@ router.patch('/:id', updateCustomer)
 router.patch('/:id/sanctions', triggerSanctions)
 router.patch('/:id/block', triggerBlock)
 router.patch('/:id/suspend', triggerSuspend)
+router.patch('/:id/dataphotos', updateIdCardData)
```
Comment (RafaelTaranto) [inline lib/routes/customerRoutes.js]: let's change the endpoint here to `/:id/photos/idcarddata`

File: lib/customers.js [hunk 8]
```diff
@@ -569,6 +570,63 @@ function updatePhotoCard (id, patch) {
...
+
+/**
+ * @param {String} id customer id
+ * @param {Object} patch customer latest id card photos
+ * @returns {Promise<Object>} new patch to be applied
+ */
+function updateIdCardData (patch, id) {
+  /* TODO: fetch operator id */
+  const operatorId = 'id-operator'
+  const directory = operatorDataDir + '/' + operatorId + '/' + id + '/'
```
Comment (RafaelTaranto) [inline lib/customers.js]:
```suggestion
  const directory = `${operatorDataDir}/${operatorId}/${id}/`
```

File: lib/customers.js [hunk 9]
```diff
@@ -569,6 +570,63 @@ function updatePhotoCard (id, patch) {
...
+      const dirname = path.join(dir)
+      console.log(dirname, 'dirnaaaaaaaaaaaaaaame')
+      // create the directory tree if needed
+      _.attempt(() => makeDir.sync(dirname))
+      const promises = imagesData.map((imageData, index) => {
+        console.log(index)
+        // decode the base64 string to binary data
+        const decodedImageData = Buffer.from(imageData, 'base64')
+
+        // i.e. ../<lamassu-server-home>/<operatorid>/<customerid>/idcarddata/1.jpg
```
Comment (RafaelTaranto) [inline lib/customers.js:593]: this comment is not related anymore.

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

File: lib/customers.js (hunk 9 above)
Comment (josepfo) [reply lib/customers.js:593]: why not?

**Review by RafaelTaranto — APPROVED** [review]

**Pull request merged**